### DISCUSSION TITLE
Extend build command to support install options.

### DISCUSSION THREAD

**Discussion by keriksson-rosenqvist-oqc** [python-poetry/poetry#7338] [Ideas]
I've got a setup that uses multiple dependency groups in my pyproject.toml file. Depending on what system I am working on I can install only certain dependencies using e.g. `poetry install --only common,frontend`. This works great but since the `build` command does not support the options `--without, --with, --only, --only-root, --extras, --all-extras` I can not easily build wheels for each sub-system. It would be great if this support could be added.

Example use: `poetry build --with common,frontend`, `poetry build --extras pyqt`

**Comment by neersighted** [comment]
I don't understand what you mean/are trying to achieve -- what packages are installed and packages are present during build/are captured in the metadata are completely independent. Are you trying to export dependency groups into the metadata of built wheels?

**Reply by keriksson-rosenqvist-oqc** [reply]
Yes exactly that. I know they are completely independent but I would like the build command to support (some of) the install option arguments as well.
I've actually pulled the repo and started looking at how it all fits together to see if it's something I could do myself, either in a forked repo or possibly add to the default. Possibly only adding the `--with` to start so that usage is backwards compatible, i.e. `poetry build` produces the same result as it currently does.

**Reply by neersighted** [reply]
Okay, that's a pretty broad feature request that has not been ruled out, but that would represent a invasive change of direction and code. I would suggest creating a feature request, if one doesn't already exist. I recall having one along the lines of 'allow exporting some groups to the main group with a flag in pyproject.toml' -- and declarative likely is the best way to get initial acceptance.

Anyway, this is definitely a case where a good description of the change you want to see, and building consensus on it, will be better than sending code first.
